cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp

```diff
@@ -0,0 +1,1230 @@
+        Float sum = Float(0);
+        Float sum2 = Float(0);
+        Float sum2cent = Float(0);
+        Float mean = Float(0);
+
+        for (std::int64_t row = 0; row < row_block_size; ++row) {
+            const std::int64_t y = (row + row_offset) * column_count;
+            const Float el = data_ptr[y + x];
+            Float inv_n = Float(1) / (row + 1);
+            Float delta = el - mean;
+
+            min = sycl::fmin(el, min);
+            max = sycl::fmax(el, max);
+            sum += el;
+            sum2 += el * el;
+            mean += delta * inv_n;
+            sum2cent += delta * (el - mean);
+        }
+
+        if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+            brc_ptr[x * row_block_count + row_block_idx] = row_block_size;
+        }
+        if constexpr (check_mask_flag(bs_list::min, List)) {
+            bmin_ptr[x * row_block_count + row_block_idx] = min;
+        }
+        if constexpr (check_mask_flag(bs_list::max, List)) {
+            bmax_ptr[x * row_block_count + row_block_idx] = max;
+        }
+        if constexpr (check_mask_flag(bs_list::sum | bs_list::mean | sum2cent_based_stat, List)) {
+            bsum_ptr[x * row_block_count + row_block_idx] = sum;
+        }
+        if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+            bsum2_ptr[x * row_block_count + row_block_idx] = sum2;
+        }
+        if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+            bsum2cent_ptr[x * row_block_count + row_block_idx] = sum2cent;
+        }
+    }
+}
+
+/* block processing kernel for device execution */
+template <typename Float, bs_list List, bool DefferedFin>
+inline void merge_blocks_kernel(sycl::nd_item<1> item,
+                                const std::int64_t* brc_ptr,
+                                const Float* bmin_ptr,
+                                const Float* bmax_ptr,
+                                const Float* bsum_ptr,
+                                const Float* bsum2_ptr,
+                                const Float* bsum2cent_ptr,
+                                std::int64_t* lrc_ptr,
+                                Float* lmin_ptr,
+                                Float* lmax_ptr,
+                                Float* lsum_ptr,
+                                Float* lsum2_ptr,
+                                Float* lsum2cent_ptr,
+                                Float* lmean_ptr,
+                                Float* rmin_ptr,
+                                Float* rmax_ptr,
+                                Float* rsum_ptr,
+                                Float* rsum2_ptr,
+                                Float* rsum2cent_ptr,
+                                Float* rmean_ptr,
+                                Float* rsorm_ptr,
+                                Float* rvarc_ptr,
+                                Float* rstdev_ptr,
+                                Float* rvart_ptr,
```

Comment:
Can you create structure to pass this data&

Reply:
I considered this option, but decided to make initial dpcpp device kernels signature similar to opencl ones as much as possible, due to i faced significant degradation in passing structures which holding pointers into kernels  during DF implementation. Therefore i will try this approach but as part of next iteration.